rules/spell-checker.js

```diff
@@ -198,26 +201,39 @@ module.exports = {
             );
         }
 
+        function underscoreParser(aNode, value, spellingType) {
+            if (options.ignoreUpperCaseUnderscore) {
+                checkSpelling(aNode, value, spellingType);
+            } else {
+                const hasUnderscore = value.split('_').length > 1;
```

Comment:
maybe we can change the order of these two sentencxes to avoid splitting the string twice. 

performance is important since we are processing a lot of strings